mixer/pkg/lang/compiler/compiler.go

```diff
@@ -522,6 +522,23 @@ func (g *generator) generateConditional(f *ast.Function, depth int, mode nilMode
 	}
 }
 
+func (g *generator) generateAdd(f *ast.Function, depth int) {
+	exprType := g.evalType(f.Args[0])
```

Comment:
you need to verify both argument types, I think

Reply:
Can you point me to the type checker addition? I see parsing applied to unspecified types and runtime checks, but no type checks.

Reply:
OK, thanks for explaining. This is a bit strange since UNSPECIFIED is really a type parameter then, not a for-all type. This might bite back in the future.